src/main/java/dk/aau/cs/io/queries/XMLQueryLoader.java

```diff
@@ -199,33 +210,68 @@ private static boolean checkQuantifiers(Node prop) {
         return true;
     }
 
+    public static boolean canBeHyperLTL(Node prop) {
+        NodeList children = prop.getChildNodes();
+        int hyperLTLNodesCount = 0;
+
+        for(int i = 0; i < children.getLength(); i++) {
+            Node child = children.item(i);
+            hyperLTLNodesCount += countHyperLTLNodes(child);
+
+            if (hyperLTLNodesCount > 0){
+                return true;
+            }
+        }
+
+        return hyperLTLNodesCount > 0;
+    }
+
+    public static int countHyperLTLNodes(Node prop) {
+        NodeList children = prop.getChildNodes();
+        int count = 0;
+        for (int i = 0; i < children.getLength(); i++) {
+            Node child = children.item(i);
+            if(child.getNodeName().equals("path-scope")) {
+                return 1;
+            }
+
+            if(child.getNodeName().equals("exists-path") || child.getNodeName().equals("all-paths")) {
+                if(child.getAttributes().getLength() > 0) {
+                    return 1;
+                }
+            }
+            count += countHyperLTLNodes(child);
+        }
+        return count;
+    }
+
     public static boolean canBeLTL(Node prop) {
         NodeList children = prop.getChildNodes();
-        int counter = 0;
+        int allPathsCounter = 0;
 
         for (int i = 0; i < children.getLength(); i++) {
             Node child = children.item(i);
             if (child.getNodeName().equals("formula")) {
-                counter += countSupportedPaths(child);
+                allPathsCounter += countAllPaths(child);
             }
         }
-        return counter == 1;
+        return allPathsCounter == 1;
     }
 
-    private static int countSupportedPaths(Node prop) {
+    private static int countAllPaths(Node prop) {
         NodeList children = prop.getChildNodes();
-        int counter = 0;
+        int allPathsCounter = 0;
 
         for (int i = 0; i < children.getLength(); i++) {
             Node child = children.item(i);
-            if (child.getNodeName().equals("all-paths") || child.getNodeName().equals("exists-path")) {
-                counter++;
-            } else if (child.getNodeName().equals("deadlock")){
+            if (child.getNodeName().equals("all-paths")) {
+                allPathsCounter++;
+            } else if (child.getNodeName().equals("exists-path") || child.getNodeName().equals("deadlock")){
                 return 100;
             }
-            counter += countSupportedPaths(child);
+            allPathsCounter += countAllPaths(child);
         }
-        return counter;
+        return allPathsCounter;
```

Comment:
There was a fix to allow exists-path for LTL queries in the CPN branch, which is not present here... The diff shows the correct code

Reply:
Fixed in https://github.com/TAPAAL/tapaal-gui/pull/39/commits/e0651b05f7e4c6329031295bd8540cab513b49f0